f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -277,20 +281,26 @@ def _populate_member_network(self, context, member, network):
         member['vxlan_vteps'] = []
         member['gre_vteps'] = []
 
-        if 'provider:network_type' in network:
-            net_type = network['provider:network_type']
-            if net_type == 'vxlan':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['vxlan_vteps'] = self._get_endpoints(
-                        context, 'vxlan', host)
-            if net_type == 'gre':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['gre_vteps'] = self._get_endpoints(
-                        context, 'gre', host)
-        if 'provider:network_type' not in network:
-            network['provider:network_type'] = 'undefined'
+        agent_config = {}
+        segment_data = self.disconnected_service.get_network_segment(
+            context, agent_config, network)
+        if segment_data:
+            network['provider:segmentation_id'] = \
+                segment_data.get('segmentation_id', None)
+            network['provider:network_type'] = \
+                segment_data.get('network_type', None)
+
+        net_type = network.get('provider:network_type', "undefined")
```

Comment:
You've removed the piece of code (see line 293) that sets provider:network_type to a default value if the key is missing from the dictionary, which I assume is necessary for the agent to process this request in a deterministic manner.